File: infobip-mobile-messaging-android-chat-sdk/src/main/java/org/infobip/mobile/messaging/chat/view/InAppChatInputFinishChecker.java
package org.infobip.mobile.messaging.chat.view;

import org.infobip.mobile.messaging.chat.core.InAppChatClient;

public class InAppChatInputFinishChecker implements Runnable {

    private final InAppChatClient chatClient;
    private String inputValue;

    public InAppChatInputFinishChecker(InAppChatClient chatClient) {
        this.chatClient = chatClient;
    }

    @Override
    public void run() {
        chatClient.sendInputDraft(inputValue);
    }

    public void setInputValue(String inputValue) {
        this.inputValue = inputValue;
    }
}

File: infobip-mobile-messaging-android-chat-sdk/src/main/java/org/infobip/mobile/messaging/chat/view/InAppChatInputFinishChecker.kt
package org.infobip.mobile.messaging.chat.view

class InAppChatInputFinishChecker(private val sendInputDraft: (String) -> Unit) : Runnable {

    private var inputValue: String? = null

    override fun run() {
        inputValue?.takeIf { it.isNotBlank() }?.let(sendInputDraft)
    }

    fun setInputValue(inputValue: String?) {
        this.inputValue = inputValue
    }

}
